server/app/controllers/applicant/UpsellController.java

```diff
@@ -150,19 +154,35 @@ public CompletionStage<Result> considerRegister(
                   request.flash().get("banner").map(m -> ToastMessage.alert(m));
 
               if (settingsManifest.getNorthStarApplicantUi(request)) {
-                NorthStarApplicantUpsellView.Params params =
-                    NorthStarApplicantUpsellView.Params.builder()
-                        .setRequest(request)
-                        .setProfile(
-                            profile.orElseThrow(
-                                () -> new MissingOptionalException(CiviFormProfile.class)))
-                        .setApplicantPersonalInfo(applicantPersonalInfo.join())
-                        .setProgramTitle(roApplicantProgramService.join().getProgramTitle())
-                        .setApplicationId(applicationId)
-                        .setMessages(messagesApi.preferred(request))
-                        .setApplicantId(applicantId)
-                        .build();
-                return ok(northStarUpsellView.render(params)).as(Http.MimeTypes.HTML);
+                if (isCommonIntake.join()) {
+                  NorthStarApplicantCommonIntakeUpsellView.Params params =
+                      NorthStarApplicantCommonIntakeUpsellView.Params.builder()
+                          .setRequest(request)
+                          .setProfile(
+                              profile.orElseThrow(
+                                  () -> new MissingOptionalException(CiviFormProfile.class)))
+                          .setApplicantPersonalInfo(applicantPersonalInfo.join())
+                          .setEligiblePrograms(maybeEligiblePrograms.orElseGet(ImmutableList::of))
```

Comment:
Non-blocking - Most the parameters in the NorthStarApplicantUpsellView.Params params are the same (request, profile, ApplicantPersonallnfo, messages, applicantId). Can only the separating factor be added under if and issued a build() there? It is easier to read if only the distinguishing factors are highlighted under "if"

Reply:
Sweet!